testcases/open_posix_testsuite/conformance/interfaces/strchr/1-1.c

```diff
@@ -0,0 +1,103 @@
+/*
+ * This file is licensed under the GPL license. For the full content
+ * of this license, see the COPYING file at the top level of this
+ * source tree.
+ */
+
+/*
+ * assertion:
+ *
+ *	The function will locate the first occurrence of c (converted to a char)
+ *	in the string pointed to by s. The terminating NUL character is
+ *	considered to be part of the string. Upon completion, the function will
+ *	return a pointer to the byte, or a null pointer if the byte was not found.
+ *
+ * method:
+ *	-Generate sample string s1
+ *	-Define one char, taking a character from the sample string(char1), one char as NULL(char2)
+	and third character out of the sample string(char3).
+ *	-Use strchr for string with char1 and store result in result1.
+ *	-Use strchr for string with char2 and store result in result2.
+ *	-Use strchr for string with char3 and store result in result3.
+ *	-Compare the result1 with pointer to the defined char(char1).
+ *	-Compare the result2 with pointer to the defined char(char2).
+ *	-Compare result3 with NULL.
+ *	-Repeat the above all steps for given number of iterations.
+
+*/
+
+#define _XOPEN_SOURCE 700
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include "posixtest.h"
+
+#define STRING_MAX_LEN 50000
+#define STEP_COUNT 2000
+#define TNAME "strchr/1-1.c"
+
+char *random_string(int len)
+{
+	int i;
+	char *output_string = malloc(len + 1);
+	if (output_string == NULL) {
+		printf(TNAME " failed to allocate memory\n");
+		exit(PTS_UNRESOLVED);
+	}
+	for (i = 0; i < len; i++) {
+		output_string[i] = rand() % 254 + 1;
+			//Avoid adding 'n' and '\0' character in the string
+			if (output_string[i] == 'n' || output_string[i] == '\0')
+				i--;
+	}
+	output_string[len] = '\0';
+	return output_string;
+}
+
+int main(void)
+{
+	int i, char_pos;
+
+	for (i = 10; i < STRING_MAX_LEN; i += STEP_COUNT) {
```

Comment:
Why `10`? Can there be some constant describing it?

Reply:
In addition to what @pevik said, why not generate the random string once, then copy subsets of that string in each iteration.

Also, why does this need to be tested more than once? Are you trying to fuzz the libcall?)

Reply:
Changed the logic, removed 10 and making use of 'for' loop counter